app/models/user_mentee_application.rb

```diff
@@ -28,9 +29,50 @@
 #
 class UserMenteeApplication < ApplicationRecord
   belongs_to :user
+  has_many :mentee_application_states, dependent: :destroy
 
   validates :available_hours_per_week, numericality: { greater_than: 0, less_than_or_equal_to: 60 }
 
   validates :city, :state, :country, :reason_for_applying, :learned_to_code, :project_experience,
     :available_hours_per_week, presence: true
+
+  after_create :create_initial_application_state
+
+  delegate :rejected?, to: :current_state
+
+  delegate :accepted?, to: :current_state
+
+  def current_status
+    mentee_application_states.last&.status&.to_sym
+  end
+
+  def current_state
+    mentee_application_states.last
+  end
+
+  def next_status
+    return if accepted? || rejected?
+    current_status_index = MenteeApplicationState.statuses.keys.index(current_status.to_sym)
+    next_status_index = current_status_index.to_i + 1
+
+    MenteeApplicationState.statuses.keys.fetch(next_status_index)
+  end
+
+  def promote_application(user)
```

Comment:
I would add a `!` to the method call, since it's a destructive action.

Reply:
you man to the promote_application method? It creates a new record